File: src/NocWMS-Client/src/app/models/warehouse-cell.ts

export enum CellStatus {
    available = 0,
    full = 1,
    fault = 2
};
export class WarehouseCell {
    constructor(warehouseCell?: Partial<WarehouseCell>) {
        if (warehouseCell) {
            this.id = warehouseCell.id;
            this.warehouseId = warehouseCell.warehouseId;
            this.cellName = warehouseCell.cellName;
            this.volume = warehouseCell.volume;
            this.remainderVolume = warehouseCell.remainderVolume;
            this.status = warehouseCell.status;
        }
    }

    public id: number;
    public warehouseId: number;
    public cellName: string;
    public volume: number;
    public remainderVolume: number;
    public status: CellStatus;
}

File: src/NocWMS-Server/Server.ApiService/Models/WarehouseCell.cs
using System.Collections.Generic;
using KiraNet.Camellia.Infrastructure.DomainModel.Data;
using Server.ApiService.Common;

namespace Server.ApiService.Models
{
    public class WarehouseCell : IEntity<int>
    {
        public int Id { get; set; }
        public int WarehouseId { get; set; }
        public string CellName { get; set; }
        public int Volume { get; set; }
        public int RemainderVolume { get; set; }
        public CellStatus Status { get; set; } = CellStatus.Available;

        public Warehouse Warehouse { get; set; }
        public List<InventoryCell> InventoryCells { get; set; }
        public List<InboundReceiptItem> InboundReceiptItems { get; set; }
        public List<OutboundReceiptItem> OutboundReceiptItems { get; set; }
    }
}
